staging/src/k8s.io/apiserver/pkg/server/options/opentelemetry.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/spf13/pflag"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"google.golang.org/grpc"
+	"k8s.io/utils/path"
+
+	"k8s.io/apiserver/pkg/opentelemetry"
+	"k8s.io/apiserver/pkg/server/egressselector"
+	"k8s.io/component-base/traces"
+)
+
+// OpenTelemetryOptions contain configuration options for opentelemetry
+// exporters
+type OpenTelemetryOptions struct {
+	// ConfigFile is the file path with api-server opentelemetry configuration.
+	ConfigFile string
+}
+
+// NewOpenTelemetryOptions creates a new instance of OpenTelemetryOptions
+func NewOpenTelemetryOptions() *OpenTelemetryOptions {
+	return &OpenTelemetryOptions{}
+}
+
+// AddFlags adds flags related to opentelemetry to the specified FlagSet
+func (o *OpenTelemetryOptions) AddFlags(fs *pflag.FlagSet) {
+	if o == nil {
+		return
+	}
+
+	fs.StringVar(&o.ConfigFile, "opentelemetry-config-file", o.ConfigFile,
+		"File with apiserver opentelemetry configuration.")
+}
+
+// Apply adds the opentelemetry settings to the global configuration.
+func (o *OpenTelemetryOptions) Apply(es *egressselector.EgressSelector) error {
+	if o == nil {
+		return nil
+	}
+
+	npConfig, err := opentelemetry.ReadOpenTelemetryConfiguration(o.ConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to read opentelemetry config: %v", err)
+	}
+	errs := opentelemetry.ValidateOpenTelemetryConfiguration(npConfig)
+	if len(errs) > 0 {
+		return fmt.Errorf("failed to validate opentelemetry configuration: %v", errs.ToAggregate())
+	}
+
+	if npConfig == nil {
+		// No config file was specified, so don't enable exporting
+		return nil
+	}
```

Comment:
npConfig is returned by `ReadOpenTelemetryConfiguration`.  I could change it so that we check to see if `o.ConfigFile == ""` instead, and move it up.  Then I would make ReadOpenTelemetryConfiguration return an error if the config file is empty.  Does that sound better?

Reply:
yeah i think that'd be a bit cleaner.

Reply:
done